### PR TITLE
Add dokku_letsencrypt library method

### LINKED ISSUE #9: letsencrypt library
We should implement a method to add/remove letsencrypt to a given repository.

### PR DESCRIPTION
**Pull request by gavinballard**
Closes #9, by adding a simple library method that can be used to enable or disable Let's Encrypt on a specific application.

This took a slightly different approach to the existing draft implementation in #49.

### CODE REVIEW

File: library/dokku_letsencrypt.py
Comment (gavinballard) [inline library/dokku_letsencrypt.py:68]: As the `dokku-letsencrypt` plugin does not (as far as I'm aware) provide a way to retrieve the current status of Let's Encrypt for a specific app, using `awk` and checking if the target app name is in the list of currently enabled apps was the only method I really had to do this.

File: library/dokku_letsencrypt.py
Comment (ltalirz) [inline library/dokku_letsencrypt.py:67]: Maybe
```suggestion
def dokku_letsencrypt_list(data):
```
?

File: library/dokku_letsencrypt.py
Comment (ltalirz) [inline library/dokku_letsencrypt.py:110]: is that correct?
your `dokku_letsencrypt` function also returns "0" when it encounters an error.

perhaps add a comment here on why it is ok to ignore that.

**Review by ltalirz — COMMENTED** [review]
thanks a lot for the contribution @gavinballard, very welcome!

It would be nice to add a test of this new functionality somewhere [here](https://github.com/dokku/ansible-dokku/blob/master/molecule/default/verify.yml) but for this we first need an app installed during the verify stage. Unfortunately, my PR #98 that would do this is blocked by a longstanding molecule bug and there doesn't seem to be any action on it... until that point no need to add a test

File: library/dokku_letsencrypt.py
Comment (gavinballard) [reply library/dokku_letsencrypt.py:67]: I was wondering about this method name, but it does only return a single value (`"0"` if Let's Encrypt is not enabled for the given app, `"1"` if it is) rather than a list. This library method is very heavily based on the existing ones (especially [dokku_proxy.py](https://github.com/discolabs/ansible-dokku/blob/e3e43f377292047a63ef08665840f49f1cd9484c/library/dokku_proxy.py), as that is also a binary on/off method, and the patterns seems to be this method essentially asking "is this enabled for the given app?"

Perhaps updating this to be `dokku_letsencrypt_enabled` and returning `True` or `False` would make sense?

**Comment by gavinballard** [comment]
Thanks for the speedy response @ltalirz!

I was keen to add a test but without an example to copy from I wasn't confident (and sounds like it's not feasible in any case given your comments on #98). This method is pretty simple and I've been using it in production, so I think it's a fair call to bring it in anyway.

Added a couple of responses to your comments, LMK what you think and I'll update accordingly.

File: library/dokku_letsencrypt.py
Comment (ltalirz) [reply library/dokku_letsencrypt.py:67]: > Perhaps updating this to be `dokku_letsencrypt_enabled` and returning `True` or `False` would make sense?

Sounds good to me.

I leave the decision up to you.

**Comment by ltalirz** [comment]
by the way, since I don't use the letsencrypt plugin, could you please comment on whether your PR also addresses https://github.com/dokku/ansible-dokku/pull/49 ?

**Comment by gavinballard** [comment]
Hey @ltalirz, I've addressed the couple of comments with a renamed method and more sensible checking.

> by the way, since I don't use the letsencrypt plugin, could you please comment on whether your PR also addresses #49 ?

Yes, that PR won't be needed once this is merged so it could be closed out.

Thanks!

**Review by ltalirz — APPROVED** [review]
thanks @gavinballard , looks good to me

**Comment by ltalirz** [comment]
I'll tag a new version once https://github.com/dokku/ansible-dokku/pull/104 is merged as well

**Pull request merged**
